core/templates/Polyfills.ts

```diff
@@ -159,3 +159,17 @@ if (navigator.mediaDevices.getUserMedia === undefined) {
     });
   };
 }
+
+// Object.entries() polyfill for Chrome 53 and below.
+if (!Object.entries) {
+  Object.entries = (obj: Object) => {
+    let ownProps = Object.keys(obj),
+      i = ownProps.length,
+      resArray = new Array(i); // Preallocate the array.
+
+    while (i--) {
+      resArray[i] = [ownProps[i], obj[ownProps[i]]];
+      return resArray;
```

Comment:
Actually, I think the variable names are not easy to understand, can you try renaming them so it is apparent what they represent? Also, let's have separate "let" statements for each of the variables.

Reply:
Done